packages/@aws-cdk/aws-ecr-assets/lib/image-asset.ts

```diff
@@ -138,6 +138,8 @@ export class DockerImageAsset extends CoreConstruct implements IAsset {
 
     // Ensure the Dockerfile is included no matter what.
     exclude.push('!' + path.basename(file));
+    // Ensure the cdk.out folder is not included to avoid infinite loops.
+    exclude.push('cdk.out');
```

Comment:
Use `Stage.of(this)?.outdir` instead of hard-coded `cdk.out`:

```suggestion
    const cdkout = Stack.of(this)?.outdir ?? 'cdk.out';
    exclude.push(cdkout);
```

Reply:
Please document this behavior in the docstring of `DockerImageAssetProps.directory`